File: Backend/models/NMK.go
package models

// NMK represents the structure of a Nasha Mukti Kendra.
type NMK struct {
	ID            string `json:"_id" bson:"_id,omitempty"`
	Name          string `json:"Name" bson:"Name"`
	Address       string `json:"Address" bson:"Address"`
	NMKCode       string `json:"NMK_Code" bson:"NMK_Code"`
	OwnerName     string `json:"Owner_Name" bson:"Owner_Name"`
	ContactNumber  int `json:"Contact_Number,omitempty" bson:"Contact_Number,omitempty"`
	Email         string `json:"Email,omitempty" bson:"Email,omitempty"`
	ImageURL      string `json:"ImageURL" bson:"ImageURL"`
	State      string `json:"State" bson:"State"`
	District      string `json:"District" bson:"District"`
	Pincode      int `json:"Pincode" bson:"Pincode"`
	IsVerified    bool `json:"IsVerified" bson:"IsVerified"`
	EstablishedYear  string `json:"Established_Year" bson:"Established_Year"`
	NMK_Image     string `json:"NMK_Image" bson:"NMK_Image"`
	NMK_Verification_Image string `json:"NMK_Verification_Image" bson:"NMK_Verification_Image"`
}

File: src/lib/models/NMK.ts
import mongoose, { Document, Schema } from 'mongoose'

export interface INMK extends Document {
  Name: string
  Address: string
  NMK_Code: string
  Owner_Name: string
  Contact_Number?: number
  Email?: string
  ImageURL: string
  State: string
  District: string
  Pincode: number
  IsVerified: boolean
  Established_Year: number
  NMK_Image: string
  NMK_Verification_Image: string
}

const NMKSchema = new Schema<INMK>(
  {
    Name: String,
    Address: String,
    NMK_Code: String,
    Owner_Name: String,
    Contact_Number: Number,
    Email: String,
    ImageURL: String,
    State: String,
    District: String,
    Pincode: Number,
    IsVerified: { type: Boolean, default: false },
    Established_Year: Number,
    NMK_Image: String,
    NMK_Verification_Image: String,
  },
  {
    collection: 'NMK',
    timestamps: false,
  }
)

export const NMK = mongoose.models.NMK || mongoose.model<INMK>('NMK', NMKSchema)
